src/FlowScene.cpp

```diff
@@ -45,6 +47,15 @@ FlowScene(std::shared_ptr<DataModelRegistry> registry,
   , _registry(std::move(registry))
 {
   setItemIndexMethod(QGraphicsScene::NoIndex);
+
+  // Note: the weirdness of using a lambda instead of a member function pointer is
+  // due to a bug in Qt 5.2.
+
+  // This connection should come first
+  connect(this, &FlowScene::connectionCreated, this, [this](Connection const& c) { setupConnectionSignals(c); });
+
+  connect(this, &FlowScene::connectionCreated, this, [this](Connection const& c) { sendConnectionCreatedToNodes(c); });
+  connect(this, &FlowScene::connectionDeleted, this, [this](Connection const& c) { sendConnectionDeletedToNodes(c); });
```

Comment:
Using a lambda instead of

```c++
connect(this, &FlowScene::connectionDeleted, this, &FlowScene::sendConnectionDeletedToNodes);
```

The latter works just fine, except the minimal version of Qt we support is 5.2, and there's a bug where it rejects this code (it has a failing static-assertion which is a false positive)